packages/core/src/RenderingEngine/BaseVolumeViewport.ts

```diff
@@ -0,0 +1,961 @@
+import { vec3 } from 'gl-matrix';
```

Comment:
I beleive the following methods are VolumeViewport specific and don't belong here in the BaseVolumeViewport 

`setOrientation, _getOrientationVectors, _getAcquisitionPlaneOrientation, _setViewPlaneToAcquisitionPlane, getBlendModes, setBlendModes, getIntensityFromWorld, setSlabThickness, getSlabThickness, canvasToWorld, worldToCanvas, getCurrentImageIdIndex, getCurrentImageId`

specially canvasToWorld and worldToCanvas requires correct camera matrices which probably doesn't work out of the box, but we don't need them right now since we don't have tools on 3DViewport. You can just console.log('needs implementation') for them

Reply:
I refactored all the methods you mentioned but `canvasToWorld` and `worldToCanvas` are actually necessary otherwise the `TrackballRotateTool` can't work with the new 3D viewport. In my tests, the original implementation seems to be working fine.